packages/core/src/modules/ledger/IndyPool.ts

```diff
@@ -0,0 +1,170 @@
+    const poolHandle = this._poolHandle
+
+    if (!poolHandle) {
+      return
+    }
+
+    this._poolHandle = undefined
+
+    // FIXME: Add type to indy-sdk
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    await this.indy.closePoolLedger(poolHandle)
+  }
+
+  public async delete() {
+    // Close the pool if currently open
+    if (this._poolHandle) {
+      await this.close()
+    }
+
+    // FIXME: Add type to indy-sdk
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    await this.indy.deletePoolLedgerConfig(this.agentConfig.poolName)
+  }
+
+  public async connect() {
+    const poolName = this.poolConfig.id
+    const genesisPath = await this.getGenesisPath()
+
+    if (!genesisPath) {
+      throw new AriesFrameworkError('Cannot connect to ledger without genesis file')
+    }
+
+    this.logger.debug(`Connecting to ledger pool '${poolName}'`, { genesisPath })
+    await this.indy.setProtocolVersion(2)
+
+    try {
+      this._poolHandle = await this.indy.openPoolLedger(poolName)
+      return this._poolHandle
+    } catch (error) {
+      if (!isIndyError(error, 'PoolLedgerNotCreatedError')) {
+        throw isIndyError(error) ? new IndySdkError(error) : error
+      }
+    }
+
+    this.logger.debug(`Pool '${poolName}' does not exist yet, creating.`, {
+      indyError: 'PoolLedgerNotCreatedError',
+    })
+    try {
+      await this.indy.createPoolLedgerConfig(poolName, { genesis_txn: genesisPath })
+      this._poolHandle = await this.indy.openPoolLedger(poolName)
+      return this._poolHandle
+    } catch (error) {
+      throw isIndyError(error) ? new IndySdkError(error) : error
+    }
+  }
+
+  private async submitRequest(request: Indy.LedgerRequest) {
+    return this.indy.submitRequest(await this.getPoolHandle(), request)
+  }
+
+  public async submitReadRequest(request: Indy.LedgerRequest) {
+    const response = await this.submitRequest(request)
+
+    if (isLedgerRejectResponse(response)) {
+      throw new LedgerError(`Ledger '${this.id}' rejected read transaction request: ${response.reason}`)
+    }
+
+    return response as Indy.LedgerReadReplyResponse
+  }
+
+  public async submitWriteRequest(request: Indy.LedgerRequest) {
+    const response = await this.submitRequest(request)
+
+    if (isLedgerRejectResponse(response)) {
+      throw new LedgerError(`Ledger '${this.id}' rejected write transaction request: ${response.reason}`)
+    }
+
+    return response as Indy.LedgerWriteReplyResponse
+  }
+
+  private async getPoolHandle() {
+    if (!this._poolHandle) {
+      return this.connect()
+    }
+
+    return this._poolHandle
+  }
+
+  private async getGenesisPath() {
+    // If the path is already provided return it
+    if (this.poolConfig.genesisPath) return this.poolConfig.genesisPath
+
+    // Determine the genesisPath
+    const genesisPath = this.fileSystem.basePath + `/afj/genesis-${this.poolConfig.id}.txn`
```

Comment:
You mean instead of `/afj/genesis-${this.poolConfig.id}.txn`? I'd like to refactor this later on (this is not new code, just moved to another file). As the fileSystem doesn't have a temp storage path yet.